iml-wasm-components/iml-action-dropdown/src/lib.rs

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+mod locks;
+
+use futures::Future;
+use locks::Locks;
+use seed::{a, attrs, button, class, div, prelude::*, spawn_local, style, Method, Request};
+use serde;
+use std::collections::HashMap;
+
+type Record = (i64, i64);
+
+#[derive(serde::Deserialize, serde::Serialize, Debug, Eq, PartialEq, Hash, Clone)]
+struct Data {
+    uuid: String,
+    records: Vec<Record>,
+}
+
+#[derive(serde::Deserialize, serde::Serialize, Clone)]
+struct MetaData {
+    limit: u32,
+    next: Option<u32>,
+    offset: u32,
+    previous: Option<u32>,
+    total_count: u32,
+}
+
+#[derive(serde::Deserialize, serde::Serialize, Clone)]
+struct AvailableActionsApiData {
+    meta: MetaData,
+    objects: Vec<AvailableAction>,
+}
+
+#[derive(serde::Deserialize, serde::Serialize, Clone)]
+struct ActionArgs {
+    host_id: u64,
+}
+
+#[derive(serde::Deserialize, serde::Serialize, Clone)]
+struct AvailableAction {
+    args: Option<ActionArgs>,
+    class_name: Option<String>,
+    confirmation: Option<String>,
+    display_group: u64,
+    display_order: u64,
+    long_description: String,
+    verb: String,
+}
+
+type AvailableActions = Vec<AvailableAction>;
+
+// Model
+
+#[derive(Clone)]
+struct Model {
+    records: Vec<Record>,
+    available_actions: Vec<AvailableAction>,
+    locks: Locks,
+    open: bool,
+}
+
+// Setup a default here, for initialization later.
+impl Default for Model {
+    fn default() -> Self {
+        Self {
+            records: vec![],
+            available_actions: vec![],
+            locks: HashMap::new(),
+            open: false,
+        }
+    }
+}
+
+// Update
+
+#[derive(Clone)]
+enum Msg {
+    Open(bool),
+    Locks(Locks),
+    AvailableActions(AvailableActionsApiData),
+}
+
+/// The sole source of updating the model; returns a fresh one.
+fn update(msg: Msg, model: Model) -> Update<Model> {
+    match msg {
+        Msg::Open(open) => Render(Model { open, ..model }),
+        Msg::Locks(locks) => Render(Model { locks, ..model }),
+        Msg::AvailableActions(AvailableActionsApiData {
```

Comment:
You don't need to define fields you don't use:

```rust
Msg::AvailableActions(AvailableActionsApiData {
            objects: available_actions,
           ..
})
```

Reply:
So interestingly, this was my first approach, but I put the `..` on the top, like this:
```
..,
objects: available_actions
```
and it gave me a syntax error. Putting the `..` on the bottom appears to work though. Thanks.